src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -660,13 +670,13 @@ module.exports.external = [
         func: function (date) {
             try {
                 var bd = date.toString();
-
-                // Should be 8 digits
-                if (!/^\d{8}$/.test(bd)) {
-                    return bd;
-                }
-
-                return bd.substring(0, 4) + '-' + bd.substring(4, 6) + '-' + bd.substring(6, 8);
+                if (!/^\d+$/.test(bd))
+                    return null;
+                if (bd.length == 6)
+                    return bd.substring(0, 4) + '-' + bd.substring(4, 6);
+                if (bd.length >= 8)
+                    return bd.substring(0, 4) + '-' + bd.substring(4, 6) + '-' + bd.substring(6, 8);
+                return bd;
```

Comment:
Since the date parsing logic is the same here and above could it be moved to a shared function? If not I think making the regex into a constant is still a good idea.

Reply:
Thanks, I can unify this part of processing logic for date and datetime, and then extract a shared function. My previous idea is to make different logic for date and datetime. For Date, if the length of digits exceeds 8, I hope to keep the first 8 digits, such as 1999010101, which will be converted to 1999-01-01, but for DateTime, it will be filled with zero to 17 digits.